apps/api/src/routes/v1_ratelimit_setOverride.ts

```diff
@@ -0,0 +1,170 @@
+import { buildUnkeyQuery } from "@unkey/rbac";
+
+const route = createRoute({
+  tags: ["ratelimit"],
+  operationId: "ratelimit.setOverride",
+  method: "post",
+  path: "/v1/ratelimit.setOverride",
+  security: [{ bearerAuth: [] }],
+  request: {
+    body: {
+      required: true,
+      content: {
+        "application/json": {
+          schema: z.object({
+            namespaceId: z.string().openapi({
+              description:
+                "Namespaces group different limits together for better analytics. You might have a namespace for your public API and one for internal tRPC routes.",
+              example: "email.outbound",
+            }),
+            identifier: z.string().openapi({
+              description:
+                "Identifier of your user, this can be their userId, an email, an ip or anything else.",
+              example: "user_123",
+            }),
+            limit: z.number().int().positive().openapi({
+              description: "How many requests may pass in a given window.",
+              example: 10,
+            }),
+            duration: z.number().int().min(1000).openapi({
+              description: "The window duration in milliseconds",
+              example: 60_000,
+            }),
+            async: z.boolean().optional().default(false).openapi({
+              description:
+                "Async will return a response immediately, lowering latency at the cost of accuracy.",
+            }),
+          }),
+        },
+      },
+    },
+  },
+  responses: {
+    200: {
+      description: "Sucessfully created a ratelimit override",
+      content: {
+        "application/json": {
+          schema: z.object({
+            success: z.boolean().openapi({
+              description:
+                "Returns true if the request should be processed, false if it was rejected.",
+              example: true,
+            }),
+            message: z.string().openapi({
+              description: "Message about the success of the operation.",
+            }),
+          }),
+        },
+      },
+    },
+    ...openApiErrorResponses,
+  },
+});
+
+export type Route = typeof route;
+export type V1RatelimitSetOverrideRequest = z.infer<
+  (typeof route.request.body.content)["application/json"]["schema"]
+>;
+export type V1RatelimitSetOverrideResponse = z.infer<
+  (typeof route.responses)[200]["content"]["application/json"]["schema"]
+>;
+
+export const registerV1RatelimitSetOverride = (app: App) =>
+  app.openapi(route, async (c) => {
+    const req = c.req.valid("json");
+    const auth = await rootKeyAuth(
+      c,
+      buildUnkeyQuery(({ or }) =>
+        or(
+          "*",
+          "ratelimit.*.create_namespace",
+          "ratelimit.*.set_override",
+          "ratelimit.*.read_override",
+          "ratelimit.*.delete_override",
+        ),
+      ),
+    );
+    // console.log(req);
+
+    const { db, analytics } = c.get("services");
+    await db.primary.transaction(async (tx) => {
```

Comment:
we need to check if the requirested namespace belongs to their workspace first. right now anyone could insert overrides into other people's namespaces by guessing the id